examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,617 @@
+}
+
+void event_handler2(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_STA_DISCONNECTED)
+    {
+        ESP_LOGI(TAG, "******** DISCONNECTED FROM AP *********");
+        esp_event_handler_unregister(WIFI_EVENT, WIFI_EVENT_STA_DISCONNECTED, &event_handler2);
+        esp_event_handler_unregister(IP_EVENT, IP_EVENT_GOT_IP6, &event_handler2);
+    }
+    else if (event_base == IP_EVENT && event_id == IP_EVENT_GOT_IP6)
+    {
+        // This is in case not only link-local address is provided
+        auto * event = static_cast<ip_event_got_ip6_t *>(event_data);
+        ESP_LOGI(TAG, "got ip6 :" IPV6STR, IPV62STR(event->ip6_info.ip));
+    }
+}
+
+void ReturnScanResults(void *)
+{
+    while (true)
+    {
+        xSemaphoreTake(s_scanresults_semaphore, portMAX_DELAY);
+        chip_rpc_ScanResults out_scan_records;
+        constexpr size_t kScanRecordsMax               = sizeof(chip_rpc_ScanResults().aps) / sizeof(chip_rpc_ScanResult);
+        uint16_t num_scan_records                      = kScanRecordsMax;
+        wifi_ap_record_t scan_records[kScanRecordsMax] = { 0 };
+        esp_err_t err                                  = esp_wifi_scan_get_ap_records(&num_scan_records, scan_records);
+        if (ESP_OK != err)
+        {
+            ESP_LOGI(TAG, "Error getting scanned APs: %d", err);
+            num_scan_records = 0;
+        }
+        out_scan_records.aps_count = num_scan_records;
+        for (size_t i = 0; i < num_scan_records; ++i)
+        {
+            memcpy(out_scan_records.aps[i].ssid.bytes, scan_records[i].ssid, sizeof(out_scan_records.aps[i].ssid.bytes));
+            out_scan_records.aps[i].ssid.size = sizeof(out_scan_records.aps[i].ssid.bytes);
+            memcpy(out_scan_records.aps[i].bssid.bytes, scan_records[i].bssid, sizeof(out_scan_records.aps[i].bssid.bytes));
+            out_scan_records.aps[i].bssid.size    = sizeof(out_scan_records.aps[i].bssid.bytes);
+            out_scan_records.aps[i].security_type = static_cast<chip_rpc_WIFI_SECURITY_TYPE>(scan_records[i].authmode);
+            out_scan_records.aps[i].channel       = scan_records[i].primary;
+            auto found_channel                    = kChannelToFreqMap.find(scan_records[i].primary);
+            out_scan_records.aps[i].frequency     = (found_channel ? found_channel->second : 0);
+            out_scan_records.aps[i].signal        = scan_records[i].rssi;
+        }
+        s_scanresult_writer.Write(out_scan_records);
+    }
+}
+
+void scan_event_handler(void * arg, esp_event_base_t event_base, int32_t event_id, void * event_data)
+{
+    if (event_base == WIFI_EVENT && event_id == WIFI_EVENT_SCAN_DONE)
+    {
+        ESP_LOGI(TAG, "******** SCAN DONE *********");
+        esp_event_handler_unregister(WIFI_EVENT, WIFI_EVENT_SCAN_DONE, &scan_event_handler);
+        xSemaphoreGive(s_scanresults_semaphore);
+    }
+}
+
+pw::Status wifi_init_sta()
```

Comment:
The naming style is changing every other function. "InitializeWifiStation" ?

Reply:
Done as part of refactor